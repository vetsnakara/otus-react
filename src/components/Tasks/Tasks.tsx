import React, { Component } from 'react'

import type { TaskListProps, TaskListItemData } from '../../interfaces'

type ListComponentInterface = React.FC<TaskListProps>

interface TasksProps {
  tasks: Array<TaskListItemData>
  ListComponent: ListComponentInterface
}

interface TasksState {
  tasks: Array<TaskListItemData>
}

class Tasks extends Component<TasksProps, TasksState> {
  private ListComponent: ListComponentInterface

  constructor(props: TasksProps) {
    super(props)

    this.ListComponent = props.ListComponent

    this.state = {
      tasks: props.tasks
    }

    this.handleToggleComplete = this.handleToggleComplete.bind(this)
    this.handleRemove = this.handleRemove.bind(this)
  }

  handleToggleComplete(id: string) {
    this.setState(({ tasks }) => ({
      tasks: tasks.map((task) =>
        task.id === id ? { ...task, completed: !task.completed } : task
      )
    }))
  }

  handleRemove(id: string) {
    this.setState(({ tasks }) => ({
      tasks: tasks.filter((task) => task.id !== id)
    }))
  }

  render() {
    const { tasks } = this.state
    const ListComponent = this.ListComponent

    if (tasks.length === 0) {
      return <p>No tasks</p>
    }

    return (
      <ListComponent
        tasks={tasks}
        onToggleComplete={this.handleToggleComplete}
        onRemove={this.handleRemove}
      />
    )
  }
}

export default Tasks
