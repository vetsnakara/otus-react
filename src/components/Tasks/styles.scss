.tasks {
  &__no-tasks-box {
    background-color: #fff;
    text-align: center;
    padding: 100px;
  }

  &__no-tasks-icon {
    & > svg {
      width: 40px;
      height: 40px;

      & > path {
        fill: #26c6da
      }
    }
  }
}