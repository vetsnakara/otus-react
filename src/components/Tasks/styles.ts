import styled from 'styled-components'
import { SVGIcon } from '../SVGIcon'

/**
 * NoTasksBox
 */
export const NoTasksBox = styled.section`
  background-color: #fff;
  text-align: center;
  padding: 100px;
`

NoTasksBox.displayName = 'NoTasksBox'

/**
 * NoTasksIcon
 */
export const NoTasksIcon = styled(SVGIcon)`
  width: 40px;
  height: 40px;
  & > path {
    fill: #26c6da;
  }
`
NoTasksIcon.displayName = 'NoTasksIcon'
