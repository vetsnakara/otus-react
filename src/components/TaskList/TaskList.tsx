import React, { FC } from 'react'

import { Task } from '../Task'

import { TaskListProps } from 'types'

import './styles.scss'

export const TaskList: FC<TaskListProps> = ({
  tasks = [],
  onToggleComplete,
  onRemove
}) => {
  if (tasks.length === 0) return null

  return (
    <ul className="task-list" data-test="task-list">
      {tasks.map((task) => {
        const { id } = task

        return (
          <li key={id} className="task-list__item" data-test="task-list-item">
            <Task
              task={task}
              onToggleComplete={() => onToggleComplete(id)}
              onRemove={() => onRemove(id)}
            />
          </li>
        )
      })}
    </ul>
  )
}
