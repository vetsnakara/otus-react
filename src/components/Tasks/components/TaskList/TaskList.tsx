import React, { FC } from 'react'

import { Task } from './components/Task'

import { TaskListProps } from 'types'

import { TaskListContainer as Container, TaskListItem as Item } from './styles'

export const TaskList: FC<TaskListProps> = ({
  tasks = [],
  onToggleComplete,
  onRemove
}) => {
  if (tasks.length === 0) return null

  return (
    <Container>
      {tasks.map((task) => {
        const { id } = task

        return (
          <Item key={id}>
            <Task
              task={task}
              onToggleComplete={() => onToggleComplete(id)}
              onRemove={() => onRemove(id)}
            />
          </Item>
        )
      })}
    </Container>
  )
}
