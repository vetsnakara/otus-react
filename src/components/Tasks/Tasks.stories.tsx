import React, { FC } from 'react'

import { Tasks } from './Tasks'
import { TaskList } from '../TaskList'

import { getTasks } from 'data'

export default {
  title: 'Tasks',
  component: Tasks
}

/**
 * With tasks
 */
export const WithTasks: FC<{}> = () => {
  const tasks = getTasks({
    titles: ['One', 'Two', 'Three']
  })

  return <Tasks tasks={tasks} ListComponent={TaskList} />
}

/**
 * Without tasks
 */
export const WithoutTasks: FC<{}> = () => {
  return <Tasks tasks={[]} ListComponent={TaskList} />
}
