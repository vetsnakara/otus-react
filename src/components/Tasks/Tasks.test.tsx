import React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import renderer from 'react-test-renderer'

import { Tasks } from './Tasks'
import { TaskList } from './components/TaskList'

import { getTasks } from 'data'

const getTasksState = (wrapper: ReactWrapper) => {
  return wrapper.state('tasks') as Array<{ completed: boolean }>
}

describe('Tasks', () => {
  test('State is correct', () => {
    const tasks = getTasks({
      titles: ['One', 'Two', 'Three']
    })

    const element = <Tasks tasks={tasks} ListComponent={TaskList} />
    const wrapper = mount(element)
    const snapshot = renderer.create(element).toJSON()

    expect(snapshot).toMatchSnapshot()
    expect(wrapper.instance().state).toEqual({ tasks })
  })

  test('NoTasksBox is rendered when no tasks passed', () => {
    const element = <Tasks tasks={[]} ListComponent={TaskList} />
    const wrapper = mount(element)
    const snapshot = renderer.create(element).toJSON()

    expect(snapshot).toMatchSnapshot()
    expect(wrapper.find('TaskList').length).toBe(0)
    expect(wrapper.find('NoTasksBox').length).toBe(1)
  })

  test('Toggle task correct', () => {
    const tasks = getTasks({
      titles: ['One', 'Two', 'Three']
    })

    const spy = jest.spyOn(Tasks.prototype, 'handleToggleComplete')

    const element = <Tasks tasks={tasks} ListComponent={TaskList} />
    const wrapper = mount(element)
    const snapshot = renderer.create(element).toJSON()

    expect(snapshot).toMatchSnapshot()

    const taskIndex = 1

    let tasksState = getTasksState(wrapper)
    expect(tasksState[taskIndex].completed).toBe(false)

    const toggleButton = wrapper
      .find('Task')
      .at(taskIndex)
      .find('ToggleButton')
      .childAt(0)

    // click to complete task
    toggleButton.simulate('click')
    expect(spy).toHaveBeenCalledTimes(1)
    expect(spy).toHaveBeenCalledWith(tasks[taskIndex].id)

    tasksState = getTasksState(wrapper)
    expect(tasksState[taskIndex].completed).toBe(true)
  })

  test('Remove task correct', () => {
    const tasks = getTasks({
      titles: ['One', 'Two', 'Three']
    })

    const spy = jest.spyOn(Tasks.prototype, 'handleRemove')

    const element = <Tasks tasks={tasks} ListComponent={TaskList} />
    const wrapper = mount(element)
    const snapshot = renderer.create(element).toJSON()

    expect(snapshot).toMatchSnapshot()

    const taskIndex = 1

    const removeButton = wrapper
      .find('Task')
      .at(taskIndex)
      .find('RemoveButton')
      .childAt(0)

    let stateTasks: Array<{ id: string }> = wrapper.state('tasks')
    expect(stateTasks.length).toBe(tasks.length)

    // click to remove
    removeButton.simulate('click')
    expect(spy).toHaveBeenCalledTimes(1)
    expect(spy).toHaveBeenCalledWith(tasks[taskIndex].id)

    // length = length - 1
    stateTasks = wrapper.state('tasks')
    expect(stateTasks.length).toBe(tasks.length - 1)

    // no task with specified id
    expect(
      stateTasks.find((task) => task.id === tasks[taskIndex].id)
    ).toBeUndefined()
  })
})
