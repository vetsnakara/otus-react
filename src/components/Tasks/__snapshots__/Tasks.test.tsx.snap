// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks NoTasksBox is rendered when no tasks passed 1`] = `
.c0 {
  background-color: #fff;
  text-align: center;
  padding: 100px;
}

.c1 {
  width: 40px;
  height: 40px;
}

.c1 > path {
  fill: #26c6da;
}

<section
  className="c0"
>
  <svg
    className="c1"
    height="100%"
    style={Object {}}
    viewBox="0 0 512 512"
    width="100%"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <path
      d="M504.502,75.496c-9.997-9.998-26.205-9.998-36.204,0L161.594,382.203L43.702,264.311c-9.997-9.998-26.205-9.997-36.204,0 c-9.998,9.997-9.998,26.205,0,36.203l135.994,135.992c9.994,9.997,26.214,9.99,36.204,0L504.502,111.7 C514.5,101.703,514.499,85.494,504.502,75.496z"
      fill="#000"
    />
  </svg>
  <h2>
    You have no tasks
  </h2>
  <p>
    Sit back and relax
  </p>
</section>
`;

exports[`Tasks Remove task correct 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px 15px;
  background-color: #fff;
}

.c4 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  width: 30px;
  height: 30px;
  border: none;
  border-radius: 50%;
  background-color: rgba(0,0,0,0.05);
  cursor: pointer;
  outline: none;
  margin-right: 10px;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c4 > svg {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c4 path {
  fill: rgba(0,0,0,0.5);
}

.c4 svg {
  opacity: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c4:hover > svg {
  opacity: 1;
}

.c7 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  width: 30px;
  height: 30px;
  border: none;
  border-radius: 50%;
  background-color: rgba(0,0,0,0.05);
  cursor: pointer;
  outline: none;
  visibility: hidden;
  margin-left: auto;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  opacity: 0;
}

.c7 > svg {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c7 path {
  fill: rgba(0,0,0,0.5);
}

.c2:hover .c7 {
  visibility: visible;
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c6 {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  margin-right: 10px;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c5 {
  width: 15px;
  height: 15px;
}

.c0 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.c1:not(:last-child) {
  margin-bottom: 5px;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <div
      className="c2 c3"
    >
      <button
        className="styles__Button-sc-11nhwp9-1 c4"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512 512"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M504.502,75.496c-9.997-9.998-26.205-9.998-36.204,0L161.594,382.203L43.702,264.311c-9.997-9.998-26.205-9.997-36.204,0 c-9.998,9.997-9.998,26.205,0,36.203l135.994,135.992c9.994,9.997,26.214,9.99,36.204,0L504.502,111.7 C514.5,101.703,514.499,85.494,504.502,75.496z"
            fill="#000"
          />
        </svg>
      </button>
      <p
        className="c6"
      >
        One
      </p>
      <button
        className="styles__Button-sc-11nhwp9-1 c7"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512.001 512.001"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M294.111,256.001L504.109,46.003c10.523-10.524,10.523-27.586,0-38.109c-10.524-10.524-27.587-10.524-38.11,0L256,217.892 L46.002,7.894c-10.524-10.524-27.586-10.524-38.109,0s-10.524,27.586,0,38.109l209.998,209.998L7.893,465.999 c-10.524,10.524-10.524,27.586,0,38.109c10.524,10.524,27.586,10.523,38.109,0L256,294.11l209.997,209.998 c10.524,10.524,27.587,10.523,38.11,0c10.523-10.524,10.523-27.586,0-38.109L294.111,256.001z"
          />
        </svg>
      </button>
    </div>
  </li>
  <li
    className="c1"
  >
    <div
      className="c2 c3"
    >
      <button
        className="styles__Button-sc-11nhwp9-1 c4"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512 512"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M504.502,75.496c-9.997-9.998-26.205-9.998-36.204,0L161.594,382.203L43.702,264.311c-9.997-9.998-26.205-9.997-36.204,0 c-9.998,9.997-9.998,26.205,0,36.203l135.994,135.992c9.994,9.997,26.214,9.99,36.204,0L504.502,111.7 C514.5,101.703,514.499,85.494,504.502,75.496z"
            fill="#000"
          />
        </svg>
      </button>
      <p
        className="c6"
      >
        Two
      </p>
      <button
        className="styles__Button-sc-11nhwp9-1 c7"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512.001 512.001"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M294.111,256.001L504.109,46.003c10.523-10.524,10.523-27.586,0-38.109c-10.524-10.524-27.587-10.524-38.11,0L256,217.892 L46.002,7.894c-10.524-10.524-27.586-10.524-38.109,0s-10.524,27.586,0,38.109l209.998,209.998L7.893,465.999 c-10.524,10.524-10.524,27.586,0,38.109c10.524,10.524,27.586,10.523,38.109,0L256,294.11l209.997,209.998 c10.524,10.524,27.587,10.523,38.11,0c10.523-10.524,10.523-27.586,0-38.109L294.111,256.001z"
          />
        </svg>
      </button>
    </div>
  </li>
  <li
    className="c1"
  >
    <div
      className="c2 c3"
    >
      <button
        className="styles__Button-sc-11nhwp9-1 c4"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512 512"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M504.502,75.496c-9.997-9.998-26.205-9.998-36.204,0L161.594,382.203L43.702,264.311c-9.997-9.998-26.205-9.997-36.204,0 c-9.998,9.997-9.998,26.205,0,36.203l135.994,135.992c9.994,9.997,26.214,9.99,36.204,0L504.502,111.7 C514.5,101.703,514.499,85.494,504.502,75.496z"
            fill="#000"
          />
        </svg>
      </button>
      <p
        className="c6"
      >
        Three
      </p>
      <button
        className="styles__Button-sc-11nhwp9-1 c7"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512.001 512.001"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M294.111,256.001L504.109,46.003c10.523-10.524,10.523-27.586,0-38.109c-10.524-10.524-27.587-10.524-38.11,0L256,217.892 L46.002,7.894c-10.524-10.524-27.586-10.524-38.109,0s-10.524,27.586,0,38.109l209.998,209.998L7.893,465.999 c-10.524,10.524-10.524,27.586,0,38.109c10.524,10.524,27.586,10.523,38.109,0L256,294.11l209.997,209.998 c10.524,10.524,27.587,10.523,38.11,0c10.523-10.524,10.523-27.586,0-38.109L294.111,256.001z"
          />
        </svg>
      </button>
    </div>
  </li>
</ul>
`;

exports[`Tasks State is correct 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px 15px;
  background-color: #fff;
}

.c4 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  width: 30px;
  height: 30px;
  border: none;
  border-radius: 50%;
  background-color: rgba(0,0,0,0.05);
  cursor: pointer;
  outline: none;
  margin-right: 10px;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c4 > svg {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c4 path {
  fill: rgba(0,0,0,0.5);
}

.c4 svg {
  opacity: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c4:hover > svg {
  opacity: 1;
}

.c7 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  width: 30px;
  height: 30px;
  border: none;
  border-radius: 50%;
  background-color: rgba(0,0,0,0.05);
  cursor: pointer;
  outline: none;
  visibility: hidden;
  margin-left: auto;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  opacity: 0;
}

.c7 > svg {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c7 path {
  fill: rgba(0,0,0,0.5);
}

.c2:hover .c7 {
  visibility: visible;
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c6 {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  margin-right: 10px;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c5 {
  width: 15px;
  height: 15px;
}

.c0 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.c1:not(:last-child) {
  margin-bottom: 5px;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <div
      className="c2 c3"
    >
      <button
        className="styles__Button-sc-11nhwp9-1 c4"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512 512"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M504.502,75.496c-9.997-9.998-26.205-9.998-36.204,0L161.594,382.203L43.702,264.311c-9.997-9.998-26.205-9.997-36.204,0 c-9.998,9.997-9.998,26.205,0,36.203l135.994,135.992c9.994,9.997,26.214,9.99,36.204,0L504.502,111.7 C514.5,101.703,514.499,85.494,504.502,75.496z"
            fill="#000"
          />
        </svg>
      </button>
      <p
        className="c6"
      >
        One
      </p>
      <button
        className="styles__Button-sc-11nhwp9-1 c7"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512.001 512.001"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M294.111,256.001L504.109,46.003c10.523-10.524,10.523-27.586,0-38.109c-10.524-10.524-27.587-10.524-38.11,0L256,217.892 L46.002,7.894c-10.524-10.524-27.586-10.524-38.109,0s-10.524,27.586,0,38.109l209.998,209.998L7.893,465.999 c-10.524,10.524-10.524,27.586,0,38.109c10.524,10.524,27.586,10.523,38.109,0L256,294.11l209.997,209.998 c10.524,10.524,27.587,10.523,38.11,0c10.523-10.524,10.523-27.586,0-38.109L294.111,256.001z"
          />
        </svg>
      </button>
    </div>
  </li>
  <li
    className="c1"
  >
    <div
      className="c2 c3"
    >
      <button
        className="styles__Button-sc-11nhwp9-1 c4"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512 512"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M504.502,75.496c-9.997-9.998-26.205-9.998-36.204,0L161.594,382.203L43.702,264.311c-9.997-9.998-26.205-9.997-36.204,0 c-9.998,9.997-9.998,26.205,0,36.203l135.994,135.992c9.994,9.997,26.214,9.99,36.204,0L504.502,111.7 C514.5,101.703,514.499,85.494,504.502,75.496z"
            fill="#000"
          />
        </svg>
      </button>
      <p
        className="c6"
      >
        Two
      </p>
      <button
        className="styles__Button-sc-11nhwp9-1 c7"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512.001 512.001"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M294.111,256.001L504.109,46.003c10.523-10.524,10.523-27.586,0-38.109c-10.524-10.524-27.587-10.524-38.11,0L256,217.892 L46.002,7.894c-10.524-10.524-27.586-10.524-38.109,0s-10.524,27.586,0,38.109l209.998,209.998L7.893,465.999 c-10.524,10.524-10.524,27.586,0,38.109c10.524,10.524,27.586,10.523,38.109,0L256,294.11l209.997,209.998 c10.524,10.524,27.587,10.523,38.11,0c10.523-10.524,10.523-27.586,0-38.109L294.111,256.001z"
          />
        </svg>
      </button>
    </div>
  </li>
  <li
    className="c1"
  >
    <div
      className="c2 c3"
    >
      <button
        className="styles__Button-sc-11nhwp9-1 c4"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512 512"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M504.502,75.496c-9.997-9.998-26.205-9.998-36.204,0L161.594,382.203L43.702,264.311c-9.997-9.998-26.205-9.997-36.204,0 c-9.998,9.997-9.998,26.205,0,36.203l135.994,135.992c9.994,9.997,26.214,9.99,36.204,0L504.502,111.7 C514.5,101.703,514.499,85.494,504.502,75.496z"
            fill="#000"
          />
        </svg>
      </button>
      <p
        className="c6"
      >
        Three
      </p>
      <button
        className="styles__Button-sc-11nhwp9-1 c7"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512.001 512.001"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M294.111,256.001L504.109,46.003c10.523-10.524,10.523-27.586,0-38.109c-10.524-10.524-27.587-10.524-38.11,0L256,217.892 L46.002,7.894c-10.524-10.524-27.586-10.524-38.109,0s-10.524,27.586,0,38.109l209.998,209.998L7.893,465.999 c-10.524,10.524-10.524,27.586,0,38.109c10.524,10.524,27.586,10.523,38.109,0L256,294.11l209.997,209.998 c10.524,10.524,27.587,10.523,38.11,0c10.523-10.524,10.523-27.586,0-38.109L294.111,256.001z"
          />
        </svg>
      </button>
    </div>
  </li>
</ul>
`;

exports[`Tasks Toggle task correct 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px 15px;
  background-color: #fff;
}

.c4 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  width: 30px;
  height: 30px;
  border: none;
  border-radius: 50%;
  background-color: rgba(0,0,0,0.05);
  cursor: pointer;
  outline: none;
  margin-right: 10px;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c4 > svg {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c4 path {
  fill: rgba(0,0,0,0.5);
}

.c4 svg {
  opacity: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c4:hover > svg {
  opacity: 1;
}

.c7 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  width: 30px;
  height: 30px;
  border: none;
  border-radius: 50%;
  background-color: rgba(0,0,0,0.05);
  cursor: pointer;
  outline: none;
  visibility: hidden;
  margin-left: auto;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  opacity: 0;
}

.c7 > svg {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c7 path {
  fill: rgba(0,0,0,0.5);
}

.c2:hover .c7 {
  visibility: visible;
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c6 {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  margin-right: 10px;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c5 {
  width: 15px;
  height: 15px;
}

.c0 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.c1:not(:last-child) {
  margin-bottom: 5px;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <div
      className="c2 c3"
    >
      <button
        className="styles__Button-sc-11nhwp9-1 c4"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512 512"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M504.502,75.496c-9.997-9.998-26.205-9.998-36.204,0L161.594,382.203L43.702,264.311c-9.997-9.998-26.205-9.997-36.204,0 c-9.998,9.997-9.998,26.205,0,36.203l135.994,135.992c9.994,9.997,26.214,9.99,36.204,0L504.502,111.7 C514.5,101.703,514.499,85.494,504.502,75.496z"
            fill="#000"
          />
        </svg>
      </button>
      <p
        className="c6"
      >
        One
      </p>
      <button
        className="styles__Button-sc-11nhwp9-1 c7"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512.001 512.001"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M294.111,256.001L504.109,46.003c10.523-10.524,10.523-27.586,0-38.109c-10.524-10.524-27.587-10.524-38.11,0L256,217.892 L46.002,7.894c-10.524-10.524-27.586-10.524-38.109,0s-10.524,27.586,0,38.109l209.998,209.998L7.893,465.999 c-10.524,10.524-10.524,27.586,0,38.109c10.524,10.524,27.586,10.523,38.109,0L256,294.11l209.997,209.998 c10.524,10.524,27.587,10.523,38.11,0c10.523-10.524,10.523-27.586,0-38.109L294.111,256.001z"
          />
        </svg>
      </button>
    </div>
  </li>
  <li
    className="c1"
  >
    <div
      className="c2 c3"
    >
      <button
        className="styles__Button-sc-11nhwp9-1 c4"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512 512"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M504.502,75.496c-9.997-9.998-26.205-9.998-36.204,0L161.594,382.203L43.702,264.311c-9.997-9.998-26.205-9.997-36.204,0 c-9.998,9.997-9.998,26.205,0,36.203l135.994,135.992c9.994,9.997,26.214,9.99,36.204,0L504.502,111.7 C514.5,101.703,514.499,85.494,504.502,75.496z"
            fill="#000"
          />
        </svg>
      </button>
      <p
        className="c6"
      >
        Two
      </p>
      <button
        className="styles__Button-sc-11nhwp9-1 c7"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512.001 512.001"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M294.111,256.001L504.109,46.003c10.523-10.524,10.523-27.586,0-38.109c-10.524-10.524-27.587-10.524-38.11,0L256,217.892 L46.002,7.894c-10.524-10.524-27.586-10.524-38.109,0s-10.524,27.586,0,38.109l209.998,209.998L7.893,465.999 c-10.524,10.524-10.524,27.586,0,38.109c10.524,10.524,27.586,10.523,38.109,0L256,294.11l209.997,209.998 c10.524,10.524,27.587,10.523,38.11,0c10.523-10.524,10.523-27.586,0-38.109L294.111,256.001z"
          />
        </svg>
      </button>
    </div>
  </li>
  <li
    className="c1"
  >
    <div
      className="c2 c3"
    >
      <button
        className="styles__Button-sc-11nhwp9-1 c4"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512 512"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M504.502,75.496c-9.997-9.998-26.205-9.998-36.204,0L161.594,382.203L43.702,264.311c-9.997-9.998-26.205-9.997-36.204,0 c-9.998,9.997-9.998,26.205,0,36.203l135.994,135.992c9.994,9.997,26.214,9.99,36.204,0L504.502,111.7 C514.5,101.703,514.499,85.494,504.502,75.496z"
            fill="#000"
          />
        </svg>
      </button>
      <p
        className="c6"
      >
        Three
      </p>
      <button
        className="styles__Button-sc-11nhwp9-1 c7"
        onClick={[Function]}
      >
        <svg
          className="c5"
          height="100%"
          style={Object {}}
          viewBox="0 0 512.001 512.001"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M294.111,256.001L504.109,46.003c10.523-10.524,10.523-27.586,0-38.109c-10.524-10.524-27.587-10.524-38.11,0L256,217.892 L46.002,7.894c-10.524-10.524-27.586-10.524-38.109,0s-10.524,27.586,0,38.109l209.998,209.998L7.893,465.999 c-10.524,10.524-10.524,27.586,0,38.109c10.524,10.524,27.586,10.523,38.109,0L256,294.11l209.997,209.998 c10.524,10.524,27.587,10.523,38.11,0c10.523-10.524,10.523-27.586,0-38.109L294.111,256.001z"
          />
        </svg>
      </button>
    </div>
  </li>
</ul>
`;
